build-logic/api-extractor/build.gradle.kts

```diff
@@ -0,0 +1,19 @@
+plugins {
+    id("gradlebuild.api-extractor")
+}
+
+description = "Extract API classes from JVM classes."
+
+dependencies {
+    // TODO These should look a lot simpler.
+    //      We should probably reuse them from build-logic-commons:build-platform,
+    //      but doing so causes a lot of Kotlin tests like ManagedPropertyJavaInterOpIntegrationTest fail
+    //      with an error about a missing method:
+    //      kotlinx.metadata.jvm.KotlinClassMetadata kotlinx.metadata.jvm.KotlinClassMetadata$Companion.read(kotlinx.metadata.jvm.KotlinClassHeader)
+    implementation("org.ow2.asm:asm:9.7")
```

Comment:
🤔 I thing we should add a comment to the `build-platform` [here](https://github.com/gradle/gradle/blob/d76b9ff8be0d7a751ecedc0be4da536b385c4028/build-logic-commons/build-platform/build.gradle.kts#L18) to let others know it would be good they update also this version.

@octylFractal I remember you very often do that, so just FYI

Reply:
Just to confirm, declaring a `api(platform("gradlebuild:build-platform"))` dependency causes kotlin failures? 

I just want to make sure thats what the comment means by `reuse them from`

Reply:
Yes. It's because build logic uses `kotlinx-metadata:0.6.0` and production code uses `0.5.0`.